pkg/utils/termination/termination.go

```diff
@@ -27,45 +27,35 @@ import (
 )
 
 // EnsureTerminated is a helper function that takes a v1.NodeClaim and calls cloudProvider.Delete() if status condition
-// on nodeClaim is not terminating. If it is terminating then it will call cloudProvider.Get() to check if the instance
-// is terminated or not. It will return an error and a boolean that indicates if the instance is terminated or not. We simply return
+// on nodeClaim is not terminating. It will return an error and a boolean that indicates if the instance is terminated or not. We simply return
 // conflict or a NotFound error if we encounter it while updating the status on nodeClaim.
 func EnsureTerminated(ctx context.Context, c client.Client, nodeClaim *v1.NodeClaim, cloudProvider cloudprovider.CloudProvider) (terminated bool, err error) {
-	// Check if the status condition on nodeClaim is Terminating
-	if !nodeClaim.StatusConditions().Get(v1.ConditionTypeInstanceTerminating).IsTrue() {
-		// If not then call Delete on cloudProvider to trigger termination and always requeue reconciliation
-		if err = cloudProvider.Delete(ctx, nodeClaim); err != nil {
-			if cloudprovider.IsNodeClaimNotFoundError(err) {
-				stored := nodeClaim.DeepCopy()
-				updateStatusConditionsForDeleting(nodeClaim)
-				// We use client.MergeFromWithOptimisticLock because patching a list with a JSON merge patch
-				// can cause races due to the fact that it fully replaces the list on a change
-				// Here, we are updating the status condition list
-				if err = c.Status().Patch(ctx, nodeClaim, client.MergeFromWithOptions(stored, client.MergeFromWithOptimisticLock{})); err != nil {
-					return false, err
-				}
-				// Instance is terminated
-				return true, nil
+	if err = cloudProvider.Delete(ctx, nodeClaim); err != nil {
+		if !nodeClaim.StatusConditions().Get(v1.ConditionTypeInstanceTerminating).IsTrue() {
```

Comment:
Do we want to add this only when we get an error? That doesn't make a ton of sense to me -- I think what we may actually want is to set this when we either get a NotFound error or we get no error (in both cases meaning that the Delete was processed or not necessary)